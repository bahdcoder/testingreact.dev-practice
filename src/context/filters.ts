import React from 'react'
import { FiltersContextInterface } from '../types/Filters'

const defaultFilterContext: FiltersContextInterface = {
  search: '',
  showingFilters: false,
  debouncedSearch: '',
  setSearch: () => {},
  toggleShowingFilters: () => {},
}

export const FiltersContext = React.createContext<FiltersContextInterface>(
  defaultFilterContext,
)
