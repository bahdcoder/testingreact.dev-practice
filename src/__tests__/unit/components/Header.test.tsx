import React from 'react'
import { render, fireEvent } from '@testing-library/react'

import Header from '../../../components/Header'
import { FiltersContext } from '../../../context/filters'

describe('The Header component', () => {
    const defaultFilterContext = {
        showingFilters: false,
        toggleShowingFilters: jest.fn()
    }

    const setupHeader = (filtersContext = defaultFilterContext) => render(
        <FiltersContext.Provider value={filtersContext as any}>
            <Header />
        </FiltersContext.Provider>
    )
    it('renders header correctly', () => {
        expect(
            setupHeader().asFragment()
        ).toMatchSnapshot()
    })

    it('toggles the filter open when the Filter button is clicked', () => {
        const { getByTestId } = setupHeader()

        const filterButton = getByTestId('FilterButton')

        fireEvent.click(filterButton)

        expect(defaultFilterContext.toggleShowingFilters).toHaveBeenCalled()
    })
})
