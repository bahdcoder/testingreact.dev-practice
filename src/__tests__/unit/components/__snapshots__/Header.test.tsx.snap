// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header component renders header correctly 1`] = `
<DocumentFragment>
  <header
    class="sc-AxhCb ccgeXB"
  >
    <span
      class="sc-AxiKw iyFHjw"
    />
    <div
      class="sc-AxjAm dGBLMY"
    >
      TESTING 
      <span />
      REACT.DEV
    </div>
    <button
      class="sc-AxirZ xKohc"
      data-testid="FilterButton"
    >
      Filter
    </button>
  </header>
</DocumentFragment>
`;
